js/component.js

```diff
@@ -0,0 +1,50 @@
+/**@type {HTMLCanvasElement} */
+
+const carImg = new Image();
+carImg.src = "/images/car.png"
+
+class Component {
+
+    constructor (x, y, w, h, ctx) {
+        this.x = x;
+        this.y = y;
+        this.w = w;
+        this.h = h;
+        this.image =  new Image ();
+        this.ctx = ctx;
+        this.speedX = 0;
+        this.speedY = 0;
+    }
+
+
+draw () {
```

Comment:
So if your using `this.image` to draw the car you need to set the `this.image.src`:

```
  draw() {
    this.image.src = "/images/car.png";
    this.ctx.drawImage(this.image, this.x, this.y, this.w, this.h);
  }
```